Intersect.Client/Interface/Game/EntityPanel/EntityBox.cs

```diff
@@ -742,14 +772,29 @@ private void UpdateMpBar(float elapsedTime, bool instant = false)
         private void UpdateXpBar(float elapsedTime, bool instant = false)
         {
             float targetExpWidth = 1;
-            if (((Player) MyEntity).GetNextLevelExperience() > 0)
-            {
-                targetExpWidth = (float) ((Player) MyEntity).Experience /
-                                 (float) ((Player) MyEntity).GetNextLevelExperience();
 
-                ExpLbl.Text = Strings.EntityBox.expval.ToString(
-                    ((Player) MyEntity)?.Experience, ((Player) MyEntity)?.GetNextLevelExperience()
+            if (((Player)MyEntity).GetNextLevelExperience() > 0)
+            {
+                targetExpWidth = (float)((Player)MyEntity).Experience /
+                                 (float)((Player)MyEntity).GetNextLevelExperience();
+                float expPercentage = targetExpWidth * 100;
+                var expPercentageText = expPercentage == 0 ? $"{expPercentage}%" : $"{expPercentage:0.00}%";
+                var expValueText = Strings.EntityBox.expval.ToString(
+                    ((Player)MyEntity)?.Experience, ((Player)MyEntity)?.GetNextLevelExperience()
                 );
+
+                if (Globals.Database.UiExpToPercentage)
+                {
+                    ExpLbl.Text = expPercentageText;
+                    ExpBackground.SetToolTipText(expValueText);
+                    ExpBar.SetToolTipText(expValueText);
+                }
+                else
+                {
+                    ExpLbl.Text = expValueText;
+                    ExpBackground.SetToolTipText(expPercentageText);
+                    ExpBar.SetToolTipText(expPercentageText);
```

Comment:
Is setting both tooltips really necessary? Is there a difference in behavior if only one of them is set?

Reply:
I was doing this because bars and their backgrounds had `"MouseInputEnabled": true` , so if i added the tooltip to the background only, the bar wouldn't allow it to popup when being over the background.
today i've just realized that having the tooltip enabled for the background is more than enough, and in order for this to work i'll have to change the assets so the actual bars over the background have `MouseInputEnabled` disabled.
I'll change this in the next commit.